kafka-persistence/src/main/scala/com/evolutiongaming/kafka/flow/kafkapersistence/KafkaPersistence.scala

```diff
@@ -0,0 +1,169 @@
+package com.evolutiongaming.kafka.flow.kafkapersistence
+
+
+import cats.data.NonEmptyList
+import cats.effect.Sync
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import cats.mtl.MonadState
+import cats.{FlatMap, Monad, data}
+import com.evolutiongaming.catshelper.{BracketThrowable, FromTry, Log}
+import com.evolutiongaming.kafka.flow.KafkaKey
+import com.evolutiongaming.kafka.flow.key.{Keys, KeysOf}
+import com.evolutiongaming.kafka.flow.persistence.{PersistenceOf, SnapshotPersistenceOf}
+import com.evolutiongaming.kafka.flow.snapshot.Snapshots.Snapshot
+import com.evolutiongaming.kafka.flow.snapshot.{SnapshotDatabase, Snapshots, SnapshotsOf}
+import com.evolutiongaming.kafka.journal.ConsRecord
+import com.evolutiongaming.kafka.journal.util.SkafkaHelper._
+import com.evolutiongaming.skafka.consumer.{ConsumerConfig, ConsumerOf, ConsumerRecord, WithSize}
+import com.evolutiongaming.skafka.producer.{Producer, ProducerRecord}
+import com.evolutiongaming.skafka.{FromBytes, Offset, Partition, ToBytes, Topic, TopicPartition}
+import com.evolutiongaming.sstream.Stream
+import com.olegpy.meow.effects._
+import scodec.bits.ByteVector
+
+import scala.concurrent.duration._
+import scala.util.control.NoStackTrace
+
+final class KafkaPersistence[F[_], K, S, A](
+                                             private[kafkapersistence] val keysOf: KeysOf[F, K],
+                                             private[kafkapersistence] val snapshots: SnapshotPersistenceOf[F, K, S, A],
+                                             private[kafkapersistence] val onRecoveryFinished: F[Unit]
+                                           )
+
+object KafkaPersistence {
+
+  private type BytesByKey = Map[String, ByteVector]
+
+  private object BytesByKey {
+    val zero: BytesByKey = Map.empty
+  }
+
+  private case class MissingOffsetError(topicPartition: TopicPartition) extends NoStackTrace
+
+  final case class Of[F[_], K, S, A](create: Partition => F[KafkaPersistence[F, K, S, A]])
```

Comment:
In other places, in Kafka Flow, these `Of` factories are created as separate traits with separate files. I have no strong opinion on how it should be (I actually thinking if we can get rid of them altogether), but it might be cool to keep the style consistent for sake of readability.

Reply:
Extracted `Of` as `KafkaPersistenceOf`